lib/Interpreter/CIFactory.cpp

```diff
@@ -893,16 +902,64 @@ namespace {
     // of the two is the underlying compiler.
 
 #ifdef __clang__
-    PPOpts.addMacroDef("__CLING__clang__=" ClingStringify(__clang__));
+  #if defined(__apple_build_version__)
+     // __apple_build_version__ >= 8000031 Xcode 8
+     #if __apple_build_version__ >= 7030029
```

Comment:
Why can we not just #define `__CLING__clang__` as `__clang__`? I don't see it as cling's responsibility to bring sanity in version macros...